source/common/mongo/proxy.cc

```diff
@@ -236,6 +260,11 @@ void ProxyFilter::onEvent(Network::ConnectionEvent event) {
       delay_timer_->disableTimer();
       delay_timer_.reset();
     }
+
+    if (drain_close_timer_) {
+      drain_close_timer_->disableTimer();
+      drain_close_timer_.reset();
```

Comment:
QQ: should timers follow RAII and cleanup any registered events in their destructors?

Reply:
They do actually do RAII cleanup. The reason this is done in various places is so that we can test for disable/destroy in the mocks. The alternative is adding an `onDestroy` method to the mock destructor, but that breaks NiceMock, so in general I have avoided that. I wish there was a better solution to this general issue but I'm not sure what it is. Open to suggestions.